source/dbo/Stored Procedures/Power BI Reporting/uspGenerateInitialSputumResults.sql

```diff
@@ -0,0 +1,79 @@
+/*
+Populate the InitialSputumSmearResult and InitialSputumPCRResult fields for each case
+Note: in all cases ‘sputum’ includes both sample types sputum - induced and sputum - spontaneous
+Calculated as follows for both smear and pcr:
+	If (any Positive or Negative results): 
+		Find results r from earliest date
+		if (r only contains negative results):
+			'Negative'
+		else:
+			'Positive'
+	Else if (any results are 'No result available'):
+		'No Result available'
+	Else if (any results 'Awaiting'):
+		'Awaiting'
+	Else (no results):
+		'No result'
+*/
+
+
+CREATE PROCEDURE [dbo].[uspGenerateInitialSputumResults]
+
+AS
+	WITH
+
+	ResultRanking AS
+	(
+		SELECT 1 AS [Rank], 1 AS [SubRank], 'Positive' AS ResultName
+		UNION
+		SELECT 1 AS [Rank], 2 AS [SubRank], 'Negative' AS ResultName
+		UNION
+		SELECT 2 AS [Rank], NULL AS [SubRank], 'NoResultAvailable' AS ResultName
+		UNION
+		SELECT 3 AS [Rank], NULL AS [SubRank], 'Awaiting' AS ResultName
+	),
+
+	NtbsInitialSputumSmearResults AS 
+	(
+		SELECT DISTINCT rr.NotificationId, rr.SourceSystem, FIRST_VALUE(mtr.[Result]) OVER (PARTITION BY rr.NotificationId ORDER BY rra.[Rank], mtr.TestDate, rra.[SubRank]) AS InitialSputumSmearResult
+		FROM [test-ntbs].[dbo].[ManualTestResult] mtr
```

Comment:
We shouldn't reference any specific databases in code, it would mean that a deployment to any environment that isn't our tests ones would fail. I think what you're looking for here is `[$(NTBS)].[dbo].[ManualTestResult]`.
`$(NTBS)` is a variable that is set by our deployment scripts to be the correct NTBS database we are deploying to. If this is confusing then let me know and we'll have a call to go through it

Reply:
I haven't ran this locally as the reporting service isn't set up for my local instance - I ended up testing it against a temporary version of Record_CaseData I made on Test, hence why I didn't need the variable